File: módulo-5/herlivre-spring/src/main/java/com/travelagency/herlivre/model/RegistroAluguelQuarto.java
package com.travelagency.herlivre.model;

import org.springframework.format.annotation.DateTimeFormat;

import javax.persistence.*;
import java.time.LocalDate;

@Entity
@Table
public class RegistroAluguelQuarto {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @ManyToOne
    @JoinColumn(name = "id_quarto",nullable = false)
    private Quarto quarto;
    @Column(nullable = false)
    @DateTimeFormat(iso = DateTimeFormat.ISO.DATE)
    private LocalDate data_entrada;
    @Column(nullable = false)
    @DateTimeFormat(iso = DateTimeFormat.ISO.DATE)
    private LocalDate data_saida;

    public RegistroAluguelQuarto(Long id, Quarto quarto, LocalDate data_entrada, LocalDate data_saida) {
        this.id = id;
        this.quarto = quarto;
        this.data_entrada = data_entrada;
        this.data_saida = data_saida;
    }

    public RegistroAluguelQuarto(){

    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Quarto getQuarto() {
        return quarto;
    }

    public void setQuarto(Quarto quarto) {
        this.quarto = quarto;
    }

    public LocalDate getData_entrada() {
        return data_entrada;
    }

    public void setData_entrada(LocalDate data_entrada) {
        this.data_entrada = data_entrada;
    }

    public LocalDate getData_saida() {
        return data_saida;
    }

    public void setData_saida(LocalDate data_saida) {
        this.data_saida = data_saida;
    }
}

File: módulo-6/herlivre-api/Herlivre_api/Models/RegistroAluguelQuarto.cs
using herlivre_api.Models;
using Microsoft.VisualBasic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Herlivre_api.Models
{
    [Table("Registro_aluguel_quarto")]
    public class RegistroAluguelQuarto
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public long Id { get; set; }

        [ForeignKey("id_quarto")]
        public Quarto Quarto { get; set; }

        public string Data_entrada { get; set; }

        public string Data_saida { get; set; }

    }
}
